readHR.py

```diff
@@ -0,0 +1,112 @@
+        self.filecheck()
+        if self.csv_check == True:
+            self.readfile()
+            self.datacheck()
+        if self.data_type == True and self.csv_check == True:
+            self.findrange()
+
+    def filecheck(self):
+        """
+            This method takes in ECG data from a CSV file inputted by the user,
+            and checks that it is the correct file format.
+
+            :return: If the file is not a CSV, the function will raise
+            "Error: File is not a .csv". For CSV files, the initialization
+            will move on to the next step.
+
+        """
+        if self.file.endswith('.csv'):
+            self.csv_check = True
+        else:
+            self.csv_check = False
+            self.time = 0
+            self.voltage = 0
+            print("Error: File is not a .csv")
+
+    def readfile(self):
+        """
+        This method reads the contents of the csv file and attempts to convert
+        numeric strings into floats before adding to time and voltage vectors.
+
+        :return: Time and voltage vectors with each entry from columns in the
+        original csv file.
+        """
+        if self.file.endswith('.csv'):
+            with open(self.file) as ecg_Data_File:
+                ecg_reader = csv.reader(ecg_Data_File)
+                next(ecg_reader)
+                for row in ecg_reader:
+                    try:
+                        u = float(row[0])
+                        self.time.append(u)
+                    except:
+                        self.time.append(row[0])
+                    try:
+                        v = float(row[1])
+                        self.voltage.append(v)
+                    except:
+                        self.voltage.append(row[1])
+
+    def datacheck(self):
+        """
+            This method takes in the time and voltage arrays and ensures that
+            the data type for every element is a float.
+
+            :param self.time: Array of time values from .CSV file assigned to class
+
+            :param self.voltage: Array of voltage values from .CSV file assigned to
+            object of class EcgReader
+
+            :return: If a particular element in time or voltage is not a float,
+            the function will raise "Error: Time vector is wrong data type".
+
+            :return: self.data_type is a 1 if numeric and 0 if any entries are
+            strings. If data_type is 0, some of the remaining initialization
+            steps will not be performed.
+
+            """
+        for i in self.time:
+            if type(i) == str:
+                self.data_type = False
+                print("Error: Time vector is wrong data type")
+                break
+        for j in self.voltage:
+            if type(j) == str:
+                self.data_type = False
+                print("Error: Voltage vector is wrong data type")
+                break
+
+    def findrange(self, peakthresh=0.9, basethresh=0.1):
```

Comment:
why are these input thresholds not keyword inputs available when initializing the object?  How would the user override when creating the object?

Reply:
Need units for these thresholds.

Reply:
So the reason I don't have this as a keyword is because these numbers basically just relate to how I made the peak detection algorithm work (looking for a voltage greater than 90% of the max found for the entire signal..). That means that 1. I didn't want users to really change this value and 2. It's unitless.
I had it in the findrange method so that if I ever had to tweak that number if the detection calculations didn't work, I could change it easily.